client/state/checklist/reducer.js

```diff
@@ -14,15 +14,25 @@ const markChecklistTaskComplete = ( state, taskId ) => ( {
 	tasks: { ...state.tasks, [ taskId ]: true },
 } );
 
+const moduleTaskMap = {
+	lazy_images: 'jetpack_lazy_images',
+	monitor: 'jetpack_monitor',
+	// Both photon and photon-cdn mark the Site Accelerator Task as enabled
+	photon: 'jetpack_site_accelerator',
+	'photon-cdn': 'jetpack_site_accelerator',
+	search: 'jetpack_search',
+	videopress: 'jetpack_video_hosting',
+};
+
 function items( state = {}, action ) {
 	switch ( action.type ) {
 		case SITE_CHECKLIST_RECEIVE:
 			return action.checklist;
 		case SITE_CHECKLIST_TASK_UPDATE:
 			return markChecklistTaskComplete( state, action.taskId );
 		case JETPACK_MODULE_ACTIVATE_SUCCESS:
```

Comment:
Do you think it makes sense to update the checklist on module disable as well? Can't really do it for Site Accelerator reliably 😞

Reply:
> Do you think it makes sense to update the checklist on module disable as well?

I was thinking about doing it in a follow-up PR :grimacing: 

> Can't really do it for Site Accelerator reliably 😞 

That's one of the reasons I wanted to tackle it separately. But yeah, if the criterion to disable it is that both `photon` and `photon-cdn` are disabled, we'd need to inspect state to get activation status for the other, right? So yeah, we can't really do that :slightly_frowning_face:

Reply:
Oh, or can we? We actually have access to the entire tasklist, so we might actually do something like

```js
		case JETPACK_MODULE_DEACTIVATE_SUCCESS:
			if (
				( action.moduleSlug === 'photon' && ! get( state.tasks, [ 'photon-cdn', 'completed' ] ) ) ||
				( action.moduleSlug === 'photon-cdn' && ! get( state.tasks, [ 'photon', 'completed' ] ) )
			) {
				return markChecklistTaskInomplete( state, 'jetpack_site_accelerator' );
			}
```

Reply:
Ugh no I'm mixing up checklist and JP module state :man_facepalming:

Reply:
Anyway.

> Do you think it makes sense to update the checklist on module disable as well?
8993de9

(For the non-Site Accelerator tasks)